src/lib/utils/inventoryUtils.ts

```diff
@@ -1,6 +1,25 @@
-import { InventoryRecord } from "../airtable/interface";
+import moment, { Moment } from 'moment';
+import { InventoryRecord, InventoryUpdateRecord, PurchaseRequestRecord } from '../airtable/interface';
+import { selectInventoryUpdatesArrayByInventoryId, selectPurchaseRequestsArrayByInventoryId } from '../redux/inventoryData';
+import { store } from '../redux/store';
 
 // Calculate when an inventory record was last updated
-export const getInventoryLastUpdated = (inventory: InventoryRecord) => {
-    return "00/00/00 00:00pm"; // TODO @wangannie: replace with real calculation
-}
\ No newline at end of file
+export const getInventoryLastUpdated = (inventory: InventoryRecord): string => {
+  const purchaseRequests: PurchaseRequestRecord[] = selectPurchaseRequestsArrayByInventoryId(store.getState(), inventory.id);
+  const inventoryUpdates: InventoryUpdateRecord[] = selectInventoryUpdatesArrayByInventoryId(store.getState(), inventory.id);
+
+  const updateTimes: Moment[] = [];
+  purchaseRequests.map((pr) => updateTimes.push(moment(pr.createdAt)));
+  inventoryUpdates.map((update) => updateTimes.push(moment(update.createdAt)));
+
+  // Check because moment will incorrectly return the current time if updateTimes is empty
+  if (updateTimes.length > 0) {
+    return moment.max(updateTimes).toLocaleString();
+  }
+  return 'Unknown'; // TODO @wangannie: address design edge case
+};
+
+// TODO @wangannie: fill in with real period calculation
+export const getCurrentPeriod = (): number => {
+  return moment().month();
```

Comment:
Just a side note (no action). Periods are defined as `YYYY-MM` because month isn't enough to uniquely identify a period

Reply:
Good catch! I forgot to delete this when I decided to remove period from Airtable